### PR TITLE
Added test dependency for official parser (issue 119)

### PR DESCRIPTION
**Pull request by Deni1593**
Changed files: root build.gradle and language/build.gradle

Added files: ParsersComparisonTest.java

### CODE REVIEW

File: language/src/test/java/parser/ParsersComparisonTest.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    Path modelPath = Path.of(MODEL_PATH, modelName);
+
+    // 1) MontiCore parser (existing behavior)
+    Optional<ASTSysMLModel> ast = SysMLv2Mill.parser().parse(modelPath.toString());
+    assertFalse(parser.hasErrors(), "MontiCore parsing should not have failed");
+    assertTrue(ast.isPresent(), "MontiCore AST should have been created");
+
+    // 2) Official OMG parser
+    String input = Files.readString(modelPath);
+    SysMLInteractiveResult result = official.eval(input);
```
Comment (adriancostin-sd) [inline language/src/test/java/parser/ParsersComparisonTest.java]: I'm not completely sure but this method probably also checks the official context conditions. So if line 70 only tests parsing then this should also only test parsing. Otherwise you are testing two different functionalities. Extract the method that is relevant and use it there. 

But first look at the other comment.

File: language/build.gradle [hunk 2]
```diff
@@ -46,6 +46,9 @@ dependencies {
 
   implementation "org.apache.commons:commons-lang3:$commons_lang_version"
   implementation "commons-cli:commons-cli:$commons_cli_version"
+
+  def omg_version = "0.18.0"
```
Comment (adriancostin-sd) [inline language/build.gradle]: 1. Move this definition in the gradle properties file and use the same version in both subprojects.

2. Please use an up to date official parser version... The java version may match but there were changes to the sysml specification since then. probably 0.58.  My fault

File: language/build.gradle [hunk 3]
```diff
@@ -46,6 +46,9 @@ dependencies {
...
+  testImplementation(group: "org.omg.sysml", name: "org.omg.sysml.interactive", version: omg_version, classifier: "all")
```
Comment (adriancostin-sd) [inline language/build.gradle]: This is a little problematic. In the pipeline we have a flag "isPipelineFullBuild" that only loads the visualization subproject only if the flag is set. This way the user can check out the projects without having a github login. This breaks that principle. This needs to be a conditional dependency and the test itself needs to be excluded from the test source sets both depending on the flag.

File: build.gradle [hunk 4]
```diff
@@ -19,6 +19,15 @@ subprojects {
       name "SE Nexus"
       url = publicNexus
     }
+
+    maven {
```
Comment (adriancostin-sd) [inline build.gradle]: This needs to be conditional as well.

**Review by adriancostin-sd — CHANGES_REQUESTED** [review]

File: language/src/omgTest/java/parser/ParsersComparisonTest.java [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+      "refinement.sysml",
+      "cardinalities.sysml",
+      "connections.sysml",
+      "collections.sysml",
+      "StateDecomposition1.sysml",
+      "FlowConectionInterfaceExample.sysml",
+      "StateActions.sysml",
+      "ConditionalSuccessionExample-1.sysml"
+  })
+  public void testParsingModels(String modelName) throws IOException {
```
Comment (adriancostin-sd) [inline language/src/omgTest/java/parser/ParsersComparisonTest.java:66]: I would also temporarely add @Disabled to the test to fix the pipeline until we decide how to fix the negative  parsing errors.
